action.yml

```diff
@@ -1,15 +1,148 @@
+
+  conflict:
+    description: How other currently in-progress runs should be handled. Values are `abort` to cancel runs in the same environment as your new run and `abort-all` to cancel all runs
+    required: false
+    default: ''
+
+  crowd:
+    description: The crowd to use for this run
+    required: false
+    default: ''
+
+  release:
+    description: Manually entered release information about the release the run is associated with
+    required: false
+    default: ''
+
+  token:
+    description: Your Rainforest QA API token
+    required: true
+
+  dry_run:
+    description: Set to true to run parameter validations without starting a new Rainforest run
+    required: false
+    default: ''
+
+  background:
+    description: Do not wait for a run to complete before exiting
+    required: false
+    default: ''
 runs:
-  using: 'docker'
-  image: 'docker://gcr.io/rf-public-images/rainforest-cli:latest'
-  args:
-    - 'run'
-    - '--fail-fast'
-    - '-f'
-    - 'spec/rainforest'
+  using: composite
+  steps:
+    - name: Set Action Version
+      shell: bash
+      run: |
+        echo "version=0.0.1" >> $GITHUB_ENV
+    - name: Validate Parameters
+      shell: bash
+      id: validate
+      run: |
+        # Show Action Version
+        echo "Using Rainforest GitHub Action v${{ env.version }}"
+
+        # Ensure results directory is there
+        mkdir -p results/rainforest
+
+        echo "::add-mask::${{ inputs.token }}"
+
+        # Validate token
+        if [ -z "${{ inputs.token }}" ] ; then
+          echo "Error: Token not set"
+          exit 1
+        fi
+
+        # Validate run_group_id
+        if ! echo "${{ inputs.run_group_id }}" | grep -Eq '^[0-9]+$' ; then
+          echo "Error: run_group_id not a positive integer (${{ inputs.run_group_id }})"
+          exit 1
+        fi
+
+        RUN_COMMAND="run --skip-update --token ${{ inputs.token }} --run-group ${{ inputs.run_group_id }} --junit-file results/rainforest/junit.xml"
+
+        # Validate conflict
+        if [ -n "${{ inputs.conflict }}" ] ; then
+          if echo "abort abort-all" | grep -Eq "\b"${{ inputs.conflict }}"\b" ; then
+            RUN_COMMAND="${RUN_COMMAND} --conflict ${{ inputs.conflict }}"
+          else
+            echo "Error: ${{ inputs.conflict }} not in (abort abort-all)"
+            exit 1
+          fi
+        fi
+
+        # Validate environment_id
+        if [ -n "${{ inputs.environment_id }}" ] ; then
+          if echo "${{ inputs.environment_id }}" | grep -Eq '^[0-9]+$' ; then
+            RUN_COMMAND="${RUN_COMMAND} --environment-id ${{ inputs.environment_id }}"
+          else
+            echo "Error: environment_id not a positive integer (${{ inputs.environment_id }})"
+            exit 1
+          fi
+        fi
+
+        # Validate crowd
+        if [ -n "${{ inputs.crowd }}" ] ; then
+          if echo "default automation automation_and_crowd on_premise_crowd" | grep -Eq "\b"${{ inputs.crowd }}"\b" ; then
```

Comment:
Prefer `case` statement?

Reply:
Assign `default automation automation_and_crowd` to a var and use here and on L104,